File: Fruit Into Baskets.js
var totalFruit = function(fruits) {
    var map = new Map();
    var max = 0;
    var right = 0;
    var left = 0;
    while(right < fruits.length){
        if(map.get(fruits[right])) map.set(fruits[right], map.get(fruits[right])+1);
        else map.set(fruits[right], 1);
        
        while(map.size > 2){
            if(map.get(fruits[left]) === 1) map.delete(fruits[left]);
            else map.set(fruits[left], map.get(fruits[left])-1);
            left++;
        }
        right++;
        max = Math.max(max, right-left);
    }
    return max
};

File: Fruit Into Baskets.py
class Solution:
    def totalFruit(self, fruits: List[int]) -> int:
        count = Counter()
        l, maxFruits = 0, 0

        for r in range(len(fruits)):
            count[fruits[r]] += 1
            while len(count) > 2:
                count[fruits[l]] -= 1
                if count[fruits[l]] == 0:
                    del count[fruits[l]]
                l += 1
            maxFruits = max(maxFruits, r - l + 1)

        return maxFruits
